test/evaluators/AggregateEvaluator.test.ts

```diff
@@ -315,38 +350,73 @@ describe('an aggregate evaluator should be able to', () => {
       })()).toEqual(undefined);
     });
 
-    describe('when we ask for throwing errors', () => {
-      it('and the input is empty', () => {
-        expect(() => {
-          testCase({
-            expr: makeAggregate('max'),
-            input: [],
-            throwError: true,
-          });
-        }).toThrow();
+    it('different types compare by string (min)', () => {
+      const result = testCase({
+        expr: makeAggregate('min'),
+        input: [
+          Bindings({ '?x': string('11') }),
+          Bindings({ '?x': int('2') }),
+          Bindings({ '?x': string('3') }),
+        ],
       });
+      expect(result).toEqual(undefined);
+    });
 
-      it('and the first value errors', () => {
-        expect(() => {
-          testCase({
-            expr: makeAggregate('max'),
-            input: [
-              Bindings({ '?x': literal('1') }),
-              Bindings({ '?x': int('1') }),
-            ],
-            throwError: true,
-          });
-        }).toThrow();
+    it('different types compare by string (max)', () => {
+      const result = testCase({
+        expr: makeAggregate('max'),
+        input: [
+          Bindings({ '?x': string('11') }),
+          Bindings({ '?x': int('2') }),
+          Bindings({ '?x': string('3') }),
+        ],
+      });
+      expect(result).toEqual(undefined);
+    });
+
+    it('different types compare by string with int type as return (max)', () => {
+      const result = testCase({
+        expr: makeAggregate('max'),
+        input: [
+          Bindings({ '?x': string('11') }),
+          Bindings({ '?x': int('2') }),
+          Bindings({ '?x': int('3') }),
+        ],
+      });
+      expect(result).toEqual(undefined);
+    });
+
+    it('passing a non-literal to max should not be accepted', () => {
+      const result = testCase({
+        expr: makeAggregate('max'),
+        input: [
+          Bindings({ '?x': nonLiteral() }),
+          Bindings({ '?x': int('2') }),
+          Bindings({ '?x': int('3') }),
+        ],
+      });
+      expect(result).toEqual(undefined);
+    });
+
+    it('passing a non-literal to sum should not be accepted', () => {
+      const result = testCase({
+        expr: makeAggregate('sum'),
+        input: [
+          Bindings({ '?x': nonLiteral() }),
+          Bindings({ '?x': int('2') }),
+          Bindings({ '?x': int('3') }),
+          Bindings({ '?x': int('4') }),
+        ],
       });
+      expect(undefined);
+    });
 
-      it('and any value in the stream errors', () => {
```

Comment:
Why was this test changed? The old version of this tests should still apply, right?